packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -103,39 +103,48 @@ export const TranslationSidebar = forwardRef<
         );
       }
     }, [word.id, notesQuery]);
+    const saveTranslatorNoteMutation = useMutation({
+      mutationFn: async (noteContent: string) =>
+        await apiClient.words.updateTranslatorNote({
+          wordId: word.id,
+          language,
+          note: noteContent,
+        }),
+    });
 
     const saveTranslatorNote = useMemo(
       () =>
         throttle(
           async (noteContent: string) => {
-            await apiClient.words.updateTranslatorNote({
-              wordId: word.id,
-              language,
-              note: noteContent,
-            });
+            await saveTranslatorNoteMutation.mutateAsync(noteContent);
             notesQuery.refetch();
           },
           15000,
           { leading: false, trailing: true }
         ),
-      [language, notesQuery, word.id]
+      [notesQuery, saveTranslatorNoteMutation]
```

Comment:
@arrocke is it safe to make this change to the dependencies array?

Reply:
`saveTranslatorNoteMutation` is going to change whenever the state of the mutation changes, we want to only rebuild this function when `mutateAsync` changes so we should destructure it before using it. Something like:
```ts
const {
  isLoading: isSavingTranslatorNote,
  mutateAsync: mutateTranslatorNote
} = useMutation()
```
We can also move `notesQuery.refetch()` into `onSuccess` of `useMutation` and then we don't need that as a dependency either

Reply:
Done.